### ISSUE TITLE
TRY201 false positive

### ISSUE THREAD

**Issue by JonathanPlasse** [charliermarsh/ruff#2164]
This code raises `TRY201`.
```python
try:
    pass
except Exception as e:
    print(e)
    raise e
```
`TRY201` wants us to simplify to:
```python
try:
    pass
except Exception:
    print(e)
    raise
```
but we use `print(e)` so the rule should not be raised.

Ruff version: `0.0.231`

**Comment by charliermarsh** [comment]
I'll fix this now.

**Comment by JonathanPlasse** [comment]
I could work on it.

**Comment by charliermarsh** [comment]
Go for it!

**Comment by JonathanPlasse** [comment]
Ok

**Comment by charliermarsh** [comment]
I think it's correct to flag this, but the `e` shouldn't be removed... Since this is valid:

```py
try:
    1/0
except Exception as e:
    print(e)
    raise
```

**Comment by charliermarsh** [comment]
Oh wait, sorry, this might not be a bug? I thought you were saying that we were trying to autofix it to the broken version.

**Comment by JonathanPlasse** [comment]
Yes, there is no bug.
